services/src/main/java/org/keycloak/authentication/authenticators/browser/ClientParameterForwardIdPAuthenticator.java

```diff
@@ -0,0 +1,124 @@
+                redirect(context, providerId);
+            }
+        } else if (context.getAuthenticatorConfig() != null && context.getAuthenticatorConfig().getConfig().containsKey(ClientParameterForwardIdPAuthenticatorFactory.DEFAULT_PROVIDER)) {
+            String defaultProvider = context.getAuthenticatorConfig().getConfig().get(ClientParameterForwardIdPAuthenticatorFactory.DEFAULT_PROVIDER);
+            LOG.tracef("Redirecting: default provider set to %s", defaultProvider);
+            storeForwardedParameters(context);
+            redirect(context, defaultProvider);
+        } else {
+        	LOG.tracef("No default provider set or %s query parameter provided", AdapterConstants.KC_IDP_HINT);
+            context.attempted();
+        }
+    }
+    
+    private void redirect(AuthenticationFlowContext context, String providerId) {
+        List<IdentityProviderModel> identityProviders = context.getRealm().getIdentityProviders();
+        for (IdentityProviderModel identityProvider : identityProviders) {
+            if (identityProvider.isEnabled() && providerId.equals(identityProvider.getAlias())) {
+                String accessCode = new ClientSessionCode<>(context.getSession(), context.getRealm(), context.getAuthenticationSession()).getOrGenerateCode();
+                String clientId = context.getAuthenticationSession().getClient().getClientId();
+                String tabId = context.getAuthenticationSession().getTabId();
+                Response response = Response.seeOther(
+                        Urls.identityProviderAuthnRequest(context.getUriInfo().getBaseUri(), providerId, context.getRealm().getName(), accessCode, clientId, tabId))
+                        .build();
+
+                LOG.debugf("Redirecting to %s", providerId);
+                context.forceChallenge(response);
+                return;
+            }
+        }
+
+        LOG.warnf("Provider not found or not enabled for realm %s", providerId);
+        context.attempted();
+    }
+
+    @Override
+    public void action(AuthenticationFlowContext context) {
+    }
+
+    @Override
+    public boolean requiresUser() {
+        return false;
+    }
+
+    @Override
+    public boolean configuredFor(KeycloakSession session, RealmModel realm, UserModel user) {
+        return true;
+    }
+
+    @Override
+    public void setRequiredActions(KeycloakSession session, RealmModel realm, UserModel user) {
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private void storeForwardedParameters(AuthenticationFlowContext context) {
+        HttpRequest httpRequest = context.getHttpRequest();
+        Map<String, String> config = context.getAuthenticatorConfig().getConfig();
+        
+        // .setNote() can only contains String. Therefore, read and put each item of forwarding parameters from loaded Properties.
+        MultivaluedMap<String, String> queryParameters = httpRequest.getUri().getQueryParameters();
```

Comment:
These parameters do not originate from client but from the authentication flow url. Client parameters are available at protocol endpoints, see e.g. https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/protocol/oidc/endpoints/AuthorizationEndpoint.java#L97